File: src/main/java/com/cuupa/dms/ui/AdressesOverview.java
package com.cuupa.dms.ui;

import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.BeforeEvent;
import com.vaadin.flow.router.HasUrlParameter;
import com.vaadin.flow.router.OptionalParameter;
import com.vaadin.flow.router.Route;

@Route(value = "adresses", layout = MainView.class)
public class AdressesOverview extends HorizontalLayout implements HasUrlParameter<String> {

    public static final String VIEW_NAME = " Adresses";

    @Override
    public void setParameter(BeforeEvent event, @OptionalParameter String parameter) {

    }
}

File: src/main/java/com/cuupa/dms/ui/AdressesOverview.kt
package com.cuupa.dms.ui

import com.cuupa.dms.ui.MainView
import com.vaadin.flow.component.orderedlayout.HorizontalLayout
import com.vaadin.flow.router.BeforeEvent
import com.vaadin.flow.router.HasUrlParameter
import com.vaadin.flow.router.OptionalParameter
import com.vaadin.flow.router.Route

@Route(value = "adresses", layout = MainView::class)
class AdressesOverview : HorizontalLayout(), HasUrlParameter<String?> {
    override fun setParameter(event: BeforeEvent, @OptionalParameter parameter: String?) {}

    companion object {
        const val VIEW_NAME = " Adresses"
    }
}